dynamodb/src/main/scala/zio/dynamodb/DynamoDBExecutorImpl.scala

```diff
@@ -173,16 +155,17 @@ private[dynamodb] final case class DynamoDBExecutorImpl private (dynamoDb: Dynam
       exclusiveStartKey = queryAll.exclusiveStartKey.map(m => attrMapToAwsAttrMap(m.map)),
       projectionExpression = toOption(queryAll.projections).map(_.mkString(", ")),
       returnConsumedCapacity = Some(buildAwsReturnConsumedCapacity(queryAll.capacity)),
-      filterExpression = maybeFilterAndKeyExpression.flatMap(_._2._1),
-      expressionAttributeValues =
-        maybeFilterAndKeyExpression.flatMap(c => aliasMapToExpressionZIOAwsAttributeValues(c._1)),
-      keyConditionExpression = maybeFilterAndKeyExpression.flatMap(_._2._2)
+      filterExpression = maybeFilterAndKeyExpr._2._1,
+      expressionAttributeValues = aliasMapToExpressionZIOAwsAttributeValues(maybeFilterAndKeyExpr._1),
+      keyConditionExpression = maybeFilterAndKeyExpr._2._2
     )
   }
 
   private def generateQueryRequest(querySome: QuerySome): QueryRequest = {
-    val maybeFilterAndKeyExpression =
-      zipOptionalRenderables(querySome.filterExpression, querySome.keyConditionExpression)
+    val maybeFilterAndKeyExpr: (AliasMap, (Option[String], Option[String])) = (for {
```

Comment:
```suggestion
    val (aliasMap, (maybeFilter, maybeKeyCond)) = (for {
```
could something like this be used?